### PR TITLE
i18n(pt-BR): Add `api-reference.md` translation

### PR DESCRIPTION
**Pull request by Yan-Thomas**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [ ] New or updated content
- [X] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

Add `api-reference.md` pt-BR translation. Special 100th contribution edition :tada:

### CODE REVIEW

File: src/pages/pt-BR/reference/api-reference.md [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+---
+let html: string = '';
+if (Astro.slots.has('default')) {
+  html = await Astro.slots.render('default')
+}
+---
+<Fragment set:html={html} />
+```
+
+`Astro.slots.render` opcinalmente aceita um segundo argumento, um array de parâmetros que serão passados para os filhos de qualquer função. Isso é extremamente útil para componentes de utilidade customizados.
```
Comment (victor0x16) [inline src/pages/pt-BR/reference/api-reference.md:165]:
```suggestion
`Astro.slots.render` opcionalmente aceita um segundo argumento, um array de parâmetros que serão passados para os filhos de qualquer função. Isso é extremamente útil para componentes de utilidade customizados.
```

File: src/pages/pt-BR/reference/api-reference.md [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+const { id } = Astro.params;
+---
+<h1>{id}</h1>
+```
+
+Então Astro irá estaticamente gerar `posts/1,`, `posts/2`, e `posts/3` em tempo de build.
+
+### Passagem de Dados com `props`
+
+Para passar dados adicionais para cada página gerada, você também pode definir um valor a `props` para cada objeto de caminho returnado. Diferente de `params`, `props` não são codificadas na URL, então não estão limitadas a apenas strings.
```
Comment (victor0x16) [inline src/pages/pt-BR/reference/api-reference.md:284]:
```suggestion
Para passar dados adicionais para cada página gerada, você também pode definir um valor a `props` para cada objeto de caminho retornado. Diferente de `params`, `props` não são codificadas na URL, então não estão limitadas a apenas strings.
```

**Review by victor0x16 — CHANGES_REQUESTED** [review]

**Comment by victor0x16** [comment]
@Yan-Thomas, great work here!!!

**Comment by Yan-Thomas** [comment]
Thanks @victor0x16!!

**Review by victor0x16 — APPROVED** [review]
LGTM

**Pull request merged**